### PR TITLE
Add proper toString() method to UpdateTask

### PR DESCRIPTION
**Pull request by ywelsch**

### CODE REVIEW

**Review by abeyad — APPROVED** [review]
LGTM, this one makes me happy :)

File: core/src/main/java/org/elasticsearch/cluster/service/ClusterService.java [hunk 1]
```diff
@@ -925,18 +925,25 @@ public TimeValue ackTimeout() {
...
-        UpdateTask(String source, T task, ClusterStateTaskConfig config, ClusterStateTaskExecutor<T> executor,
-                   ClusterStateTaskListener listener) {
-            super(config.priority(), source);
+        UpdateTask(String source, T task, Priority priority, ClusterStateTaskExecutor<T> executor, ClusterStateTaskListener listener) {
+            super(priority, source);
             this.task = task;
-            this.config = config;
             this.executor = executor;
             this.listener = listener;
+            this.description = createToString(executor.describeTasks(Collections.singletonList(task)));
```
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/cluster/service/ClusterService.java]: Is it necessary to eagerly create the description?

**Review by jasontedor — COMMENTED** [review]
I left a question.

**Review by jasontedor — APPROVED** [review]
LGTM.

**Pull request merged**